Tests/SwiftSyntaxTest/BumpPtrAllocatorTests.swift

```diff
@@ -0,0 +1,43 @@
+#if DEBUG
+// 'BumpPtrAllocator' is not a public API. Test it only when the module is built
+// with @testable (i.e. `DEBUG`)
+
+import XCTest
+@testable import SwiftSyntax
+
+final class BumpPtrAllocatorTests: XCTestCase {
+
+    func testBasic() throws {
+      let allocator = BumpPtrAllocator()
+
+      let byteBuffer = allocator.allocate(byteCount: 42, alignment: 4)
+      XCTAssertNotNil(byteBuffer.baseAddress)
+      XCTAssertEqual(byteBuffer.count, 42)
+      XCTAssertEqual(byteBuffer.baseAddress?.alignedUp(toMultipleOf: 4),
+                     byteBuffer.baseAddress)
+
+      let emptyBuffer = allocator.allocate(byteCount: 0, alignment: 8)
+      XCTAssertNil(emptyBuffer.baseAddress)
+      XCTAssertEqual(emptyBuffer.count, 0)
+
+      let typedBuffer = allocator.allocate(UInt64.self, count: 12)
+      XCTAssertNotNil(typedBuffer.baseAddress)
+      XCTAssertEqual(typedBuffer.count, 12)
+      XCTAssertEqual(UnsafeRawBufferPointer(typedBuffer).count,
+                     MemoryLayout<UInt64>.stride * 12)
+      XCTAssertEqual(byteBuffer.baseAddress?.alignedUp(toMultipleOf: MemoryLayout<UInt64>.alignment),
+                     byteBuffer.baseAddress)
+
+      let typedEmptyBuffer = allocator.allocate(String.self, count: 0)
+      XCTAssertNil(typedEmptyBuffer.baseAddress)
+      XCTAssertEqual(typedEmptyBuffer.count, 0)
+
+      XCTAssertTrue(allocator.contains(address: typedBuffer.baseAddress!))
+      XCTAssertTrue(allocator.contains(address: typedBuffer.baseAddress!.advanced(by: typedBuffer.count)))
+
+      XCTAssertEqual(allocator.totalSizeAllocated,
+                     42 + MemoryLayout<UInt64>.stride * 12)
+    }
```

Comment:
Do you think it’s worth adding a test case that shows how to allocate a Swift class on the bump allocator and pass it around? I think it would be nice to have a reduced example how to use the bump allocator in Swift.

Reply:
You should not use memory allocated by `BumpPtrAllocator` to place Swift `class` because it doesn't deinitialize them automatically. You should only place strictly trivial types i.e. no classes, closures, existentials, indirect enums, etc. (I will add doc-comments about it)

I'm not sure what you are suggesting here. The usage of the allocated memory is the same as Swift's `UnsafeXXXPointer`, the only difference is that you can't call `deallocate()` to them.

Reply:
OK, I think I meant a `struct` instead. I think I’m just not very familiar with the `UnsafeXXXPointer` APIs. I’m fine to also keep this out of this PR and be surprised by the next PR in which you start using the bump allocator.